### PR TITLE
FIX Nicely truncate link title/URL

### PR DESCRIPTION
**Pull request by GuySartorelli**
Works regardless of screen size - check with split view and in edit view.

Also check with "DRAFT", "MODIFIED", and published (no) versioned labels.

Before:
linkfield on 4 branch without this PR

After:
linkfield with this PR

## Issue
- https://github.com/silverstripe/silverstripe-linkfield/issues/138

### CODE REVIEW

File: client/src/components/LinkPicker/LinkPicker.scss [hunk 1]
```diff
@@ -106,10 +106,13 @@
   align-items: center;
   flex-grow: 1;
   height: 100%;
+  min-width: 0;
   text-align: left;
   border: none;
   margin-right: 0;
-  &::before {
+
+  &[class*="font-icon-"]::before {
```
Comment (ChloeSartorelli) [inline client/src/components/LinkPicker/LinkPicker.scss:114]: Required to override existing styles due to specificity

**Review by GuySartorelli — COMMENTED** [review]
I ended up using https://css-tricks.com/flexbox-truncated-text/ as a guide for how to do this.

**Review by emteknetnz — CHANGES_REQUESTED** [review]
Versioned badge spacing is messed up in firefox and chrome

Before:
image

After:
image

**Comment by GuySartorelli** [comment]
By "spacing" I assume you mean the vertical alignment of the text, and not any horizontal padding or anything else?

**Comment by GuySartorelli** [comment]
Fixed vertical alignment of versioned badge

**Review by emteknetnz — APPROVED** [review]
Tested locally, works good

**Pull request merged**